auditor/constraint-registry.go

```diff
@@ -33,22 +33,32 @@ func (i *ConstrainInfo) ToOpaConstraint() *unstructured.Unstructured {
 	return &constraint
 }
 
-type ConstraintRegistry map[string]*ConstrainInfo
+type ConstraintRegistry struct {
+	constraintsInfo map[string]*ConstrainInfo
+	templatesInfo   map[string]int
+}
 
 func NewConstraintRegistry() ConstraintRegistry {
-	return make(map[string]*ConstrainInfo)
+	return ConstraintRegistry{
+		constraintsInfo: make(map[string]*ConstrainInfo),
+		templatesInfo:   make(map[string]int),
+	}
 }
 
 func (r ConstraintRegistry) RegisterConstraint(c *agent.Constraint) error {
```

Comment:
All methods on ContainerRegistry should have a pointer receiver as the ContainerRegistry is now a struct and no longer an alias of a  reference type (map)